esp-wifi/src/lib.rs

```diff
@@ -181,10 +186,29 @@ impl EspWifiInitialization {
     }
 }
 
+#[derive(Debug, PartialEq, PartialOrd)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[non_exhaustive]
+/// An internal struct designed to make [`EspWifiDeinitialization`] uncreatable
+/// outside of this crate.
+pub struct EspWifiDeinitializationInternal;
+
+#[derive(Debug, PartialEq, PartialOrd)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+/// Represents the state of being deinitialized for WiFi, Bluetooth or both.
+pub enum EspWifiDeinitialization {
```

Comment:
I don't get why this is needed. Just transform `EspWifiInitialization` if something changes.